lib/rucio/core/trace.py

```diff
@@ -329,12 +352,17 @@ def validate_schema(obj):
 
     :param obj: The object to validate.
 
-    :raises: ValidationError
+    :raises: InvalidObject
     """
     obj = json.loads(obj)
 
     try:
         if obj and 'eventType' in obj:
+            event_type = SCHEMAS.get(obj['eventType'].lower())
+            if not event_type:
+                raise TypeError(f"Trace schema for eventType {obj['eventType']} is not currently supported.")
```

Comment:
I am wondering why we should differentiate here and raise the `TypeError` and not instead also go for the `InvalidObject`, which will then be handled here
https://github.com/rucio/rucio/blob/c9ea1c2dcb2a64f9e2ccac49d021914b863a684e/lib/rucio/core/trace.py#L296

In the end, it's just a schema validation error, just that in this case the schema does not exists, while in other cases it does not match attributes.

Reply:
TypeError should only be used for Python types IMHO, ValidationError makes more sense here.

Reply:
@bari12, since the idea is to reduce log size and just print an error or warning instead of the stack traces when a schema is missing in 
https://github.com/rucio/rucio/blob/c9ea1c2dcb2a64f9e2ccac49d021914b863a684e/lib/rucio/core/trace.py#L227-L241
I raise and handle the `TypeError` as the `ValidationError` currently just raises an InvalidObject error and prints the stack trace, which defeats the purpose of reducing the logs